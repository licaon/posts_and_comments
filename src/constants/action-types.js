export const INIT = 'INIT';

export const START_LOADING_DATA = 'START_LOADING_DATA';
export const END_LOADING_DATA = 'END_LOADING_DATA';

export const GET_POSTS = 'GET_POSTS';
export const ADD_POST = 'ADD_POST';
export const DELETE_POST = 'DELETE_POST';

export const ADD_SEARCH_FILTER = 'ADD_SEARCH_FILTER';
export const ADD_SELECT_FILTER = 'ADD_SELECT_FILTER';

export const GET_USERS = 'GET_USERS';
export const ADD_USER = 'ADD_USER';
