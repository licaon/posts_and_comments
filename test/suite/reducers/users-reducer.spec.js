import {  ADD_USER } from 'src/constants/action-types';
import UsersReducer from 'src/reducers/users-reducer';

describe('UsersReducer', () => {
  it(`should update reducer correctly on ${ADD_USER}`, (done) => {
    expect(UsersReducer(
      undefined,
      {
        type: ADD_USER,
        payload: {
          user: 1,
        },
      },
    )).toMatchSnapshot();
    done();
  });
  it(`should return reducer correctly on unknown action`, (done) => {
    expect(UsersReducer(
      undefined,
      { type: 'unknown' },
    )).toMatchSnapshot();
    done();
  });
});
