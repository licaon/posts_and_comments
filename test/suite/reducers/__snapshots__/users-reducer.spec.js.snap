// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersReducer should return reducer correctly on unknown action 1`] = `
Object {
  "users": Array [],
}
`;

exports[`UsersReducer should update reducer correctly on ADD_USER 1`] = `
Object {
  "users": Array [
    Object {
      "user": 1,
    },
  ],
}
`;
